src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,221 @@
+    private int size;
+
     @Override
     public void add(T value) {
+        if (head == null && tail == null && size == 0) {
+            addToEmptyList(value);
+        } else if (head != null && tail == null) {
+            addInside(value);
+        } else {
+            addIfLast(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index is bigger than size");
+        }
+        if (size == 0) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            addAtFirstPosition(value);
+            return;
+        } else if (index == size) {
+            addOnLastPosition(value);
+            return;
+        }
+        addByIndex(index, value);
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getElementOnIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T oldValue = getElementOnIndex(index).value;
+        getElementOnIndex(index).value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        final Node<T> current = getElementOnIndex(index);
+        if (size == 1) {
+            head = null;
+            tail = null;
+            size--;
+            return null;
+        }
+        Node<T> tempPrevious = getElementOnIndex(index).previous;
+        Node<T> tempNext = getElementOnIndex(index).next;
+        if (getElementOnIndex(index).previous == null) {
+            tempNext.previous = null;
+            head = tempNext;
+            size--;
+            return current.value;
+        } else if (getElementOnIndex(index).next == null) {
+            tempPrevious.next = null;
+            tail = tempPrevious;
+            size--;
+            return current.value;
+        } else {
+            tempPrevious.next = tempNext;
+            tempNext.previous = tempPrevious;
+            size--;
+            return current.value;
+        }
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        if (size == 1) {
+            head = null;
+            size--;
+            return true;
+        }
+        if (head.value.equals(object)) {
+            head = head.next;
+            size--;
+            return true;
+        }
+        Node<T> current = head;
+        int index = 0;
+        while (!Objects.equals(current.value, object)) {
+            current = current.next;
+            index++;
+            if (current == null) {
+                break;
+            }
+        }
+        if (index < size) {
+            remove(index);
+            return true;
+        } else {
+            return false;
+        }
```

Comment:
very overengineered + you can't use objects.equals here yet
```suggestion
        Node<T> node = head;
        while (node != null) {
            if (node.value == t || node.value != null && node.value.equals(t)) {
                unlink(node);
                return true;
            }
            node = node.next;
        }
        return false;
```

Reply:
you should create  `unlink` method as mentioned in the common mistakes checklist, and use it in remove methods